.github/workflows/build-with-containers.yml

```diff
@@ -0,0 +1,123 @@
+name: Build
+on:
+  push:
+    branches: main
+  pull_request:
+    branches: main
+
+permissions:
+  contents: write
```

Comment:
Does this still need write permissions, or would read be enough?

Reply:
Good catch, you're correct